python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -340,8 +327,33 @@ def _messages_thread(
                     break
             time.sleep(1)
 
+    def _logs_thread(
+        self,
+        params: PipesParams,
+        is_task_complete: Event,
+    ) -> None:
+        # Start once we have received the opened message from the external process
+        while self.opened_payload is None:
+            time.sleep(0.1)
```

Comment:
so this will spin forever if we never receive the open payload? That seems fragile. Some sort of timeout seems necessary.